### PR TITLE
citations check

### PR DESCRIPTION
**Pull request by wandmagic**
# Committer Notes

add citations check

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [x] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Comment by wandmagic** [comment]
https://github.com/GSA/fedramp-automation/issues/815

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -127,11 +127,21 @@
     <context>
         <metapath target="/system-security-plan/back-matter"/>
         <constraints>
+            <expect id="fedramp-citations-has-correct-link" target="resource[prop[@name='type' and @value='citation' and @class='fedramp-citations']]" test="count(rlink[@href = 'https://www.fedramp.gov/assets/resources/templates/FedRAMP-Laws-Regulations-Standards-and-Guidance-Reference.xlsx']) eq 1" level="ERROR">
+                <formal-name>FedRAMP Citations Has Correct Link</formal-name>
+                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/5-attachments/"/>
```
Comment (DimitriZhurkin) [inline src/validations/constraints/fedramp-external-constraints.xml:132]: For `help-url`, would that page be better?
https://automate.fedramp.gov/documentation/general-concepts/oscal-citations-and-attachments/

Just a comment. Feel free to ignore.

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 2]
```diff
@@ -127,11 +127,21 @@
...
+            </expect>
             <expect id="has-configuration-management-plan" target="." test="exists(resource[prop[@name eq 'type' and @value eq 'plan' and @class eq 'configuration-management-plan']])" level="ERROR">
                 <formal-name>Has Configuration Management Plan</formal-name>
                 <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/5-attachments/"/>
                 <message>A FedRAMP SSP MUST have a Configuration Management Plan attached.</message>
             </expect>
+            <expect id="has-fedramp-citations" target="." test="count(resource[prop[@name='type' and @value='citation' and @class='fedramp-citations']]) = 1" level="ERROR">
+                <formal-name>Has FedRAMP Citations Reference</formal-name>
+                <prop namespace="https://docs.oasis-open.org/sarif/sarif/v2.1.0" name="help-url" value="https://automate.fedramp.gov/documentation/ssp/5-attachments/"/>
```
[the remark this answers is not in the document]
Comment (wandmagic) [reply src/validations/constraints/fedramp-external-constraints.xml]: added

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
Comment (wandmagic) [reply src/validations/constraints/fedramp-external-constraints.xml]: added

**Review by DimitriZhurkin — APPROVED** [review]

**Review by Rene2mt — APPROVED** [review]
Looks good!

**Pull request merged**